pkg/controllers/networking/pod_controller.go

```diff
@@ -183,61 +184,57 @@ func (r *PodReconciler) selectNetwork(pod *corev1.Pod) (string, error) {
 	switch networkType {
 	case types.Underlay:
 		// try to get underlay network by node indexer
-		var networkList *networkingv1.NetworkList
-		var err error
-		if networkList, err = utils.ListNetworks(r, client.MatchingFields{IndexerFieldNode: pod.Spec.NodeName}); err != nil {
-			return "", fmt.Errorf("unable to list underlay network by indexer node: %v", err)
-		}
-		if len(networkList.Items) >= 1 {
-			return networkList.Items[0].GetName(), nil
+		underlayNetworkName, err := r.getNetworkByNodeNameIndexer(pod.Spec.NodeName)
+		if err != nil {
+			return "", fmt.Errorf("unable to get underlay network by node name indexer: %v", err)
 		}
 
-		// fall back to find underlay network by label selector
-		var underlayNetworkName string
-		if underlayNetworkName, err = utils.FindUnderlayNetworkForNodeName(r, pod.Spec.NodeName); err != nil {
-			return "", fmt.Errorf("unable to find underlay network for node %s", pod.Spec.NodeName)
-		}
-		if len(underlayNetworkName) == 0 {
-			return "", fmt.Errorf("no underlay network match node %s", pod.Spec.NodeName)
-		}
-		if !r.matchNetworkTypeInManager(underlayNetworkName, types.Underlay) {
```

Comment:
`matchNetworkTypeInManager` is helpful here because network cache in IPAM is syncing asynchronously, and this will disallow ip allocation if network type mismatch, which happened on a quick pod creation after network type changing.

Reply:
It seems that changing network type is not supported (and will not be supported) in validating webhook.

Reply:
yes, network type is unchangeable now, but delete and recreate is equal to "change" in an asynchronous system, I believe it's better to keep this logic.

Reply:
Done.